runtime/runtime/src/balance_checker.rs

```diff
@@ -156,6 +157,38 @@ pub(crate) fn check_balance(
             ..final_delayed_receipt_indices.next_available_index,
     )?;
 
+    // Buffered receipts
```

Comment:
This looked scary for a second :) 

nit: This method is huge can you move it to a helper method? Or perhaps follow up and refactor the whole thing into something manageable.